torchdynamo/variables/tensor.py

```diff
@@ -671,6 +676,17 @@ def from_tensor_variable(cls, tensor_variable, raw_value, need_unwrap=True):
             need_unwrap=need_unwrap,
         )
 
+    def convert_to_constant(self, tx):
+        for graph_arg in tx.output.graphargs:
+            if graph_arg.source is self.source:
+                graph_arg.erase()
+
+        for g in self.guards:
+            if g is self.volatile_guard:
```

Comment:
what are volatile guards?

Reply:
```volatile_guard``` is a mark to identify this guard may be changed later, I use it when initializing ```UnspecializedPythonVariable``` coz it's possible to convert to ```ConstantVariable``` when it encounter some operations, e.g, control flow, list index, etc.

Reply:
See my comment below - I think this system is too complicated for ultimately just removing a single type of guard `TYPE_MATCH`. Let's just remove all the volatile guard notions and explicitly remove the `TYPE_MATCH` and replace it with `CONSTANT_MATCH ` in this flow?

Reply:
See this https://github.com/pytorch/torchdynamo/pull/714#discussion_r939038623